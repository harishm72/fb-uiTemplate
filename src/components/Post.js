import React , {Component} from 'react';
import LikeButton from './LikeButton';
import CommentComponent from './CommentComponent';

class Post extends Component{
    render(){
        let post = this.props.post
        return(
            <div className="post-item">
            <p>{JSON.stringify(post["item_description"])}</p>
            {post['image'] ? <img src={post['image']} alt="" /> : null } 
            <LikeButton likes={post.likes} isLiked={post.isLiked} id={post.id}/>
            <CommentComponent comments={(post['comments'])} id={post.id}/>
            </div>
        )
    }
}

export default Post;