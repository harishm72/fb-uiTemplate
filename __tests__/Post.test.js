import React from 'react';
import { shallow } from 'enzyme';
import Post from '../src/components/Post';
import renderer from 'react-test-renderer'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

const post = {
    "id": 1,
    "isLiked": false,
    "item_description": "Declare variables not wars!!",
    "image": "",
    "likes": 10,
    "comments": [{
        "comment": "Well said, i agree.",
        "created_at": "Fri May 11 2018 19:17:43 GMT+0530 (IST)"
    },
    {
        "comment": "Good one!",
        "created_at": "Fri May 10 2018 15:23:36 GMT+0530 (IST)"
    }
    ]
}

describe('>>>Post --- Snapshot', () => {
    it('Rednders the Post Component', () => {
        const renderedValue = renderer.create(<Post post={post} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('>>>Post --- Shallow Render REACT COMPONENTS', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallow(<Post post={post} />)
    })

    it('+++ render the DUMB component', () => {
        expect(wrapper.length).toEqual(1)
    });

    it('has a "p" tag with item description', () => {
        expect(wrapper.contains(<p>{post['item_description']}</p>)).toBe(true)
    })

    it('has a "p" tag with value of item description', () => {
        expect(wrapper.find('p').get(0).props.children).toBe(post['item_description'])
    })

});