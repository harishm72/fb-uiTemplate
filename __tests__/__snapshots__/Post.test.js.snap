// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Post --- Snapshot Rednders the Post Component 1`] = `
<div
  className="post-item"
>
  <p>
    Declare variables not wars!!
  </p>
  <div>
    <p>
      10
       Likes
    </p>
    <button
      className="like-button"
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <div>
    <h3>
      Comments..
    </h3>
    <div
      className="comment"
    >
      <p>
        Well said, i agree.
      </p>
      <p
        className="comment-time"
      >
        Created at : 
        Fri May 11 2018 19:17:43 GMT+0530 (IST)
      </p>
    </div>
    <div
      className="comment"
    >
      <p>
        Good one!
      </p>
      <p
        className="comment-time"
      >
        Created at : 
        Fri May 10 2018 15:23:36 GMT+0530 (IST)
      </p>
    </div>
    <div
      className="new-comment"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="comment-input"
          onChange={[Function]}
          placeholder="type a comment here...."
          value=""
        />
      </form>
    </div>
  </div>
</div>
`;
